managers/bosh-manager/BoshTaskStatusPoller.js

```diff
@@ -75,26 +119,8 @@ class BoshTaskStatusPoller extends BaseStatusPoller {
               error: utils.buildErrorJson(err)
             }
           });
-        }
-      })
-      .catch(AssertionError, err => {
-        logger.error(`Error occured while getting last operation for instance ${instanceId}`, err);
-        this.clearPoller(instanceId, intervalId);
-        lastOperationOfInstance = {
-          state: CONST.APISERVER.RESOURCE_STATE.FAILED,
-          description: CONST.SERVICE_BROKER_ERR_MSG
-        };
-        return eventmesh.apiServerClient.updateResource({
-          resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
-          resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
-          resourceId: instanceId,
-          status: {
-            lastOperation: lastOperationOfInstance,
-            state: CONST.APISERVER.RESOURCE_STATE.FAILED,
-            error: utils.buildErrorJson(err)
-          }
-        });
-      })
+        })
+      )
       .finally(() => {
```

Comment:
`finally` block can also be put in same promise chain of `directorService`

Reply:
By previous code flow, `finally` block will still be executed if `directorService.lastOperation` fails. If `finally` is moved within same promise chain, it's execution will depend `directorService.lastOperation` being successfully resolved. @sauravmndl could you pls confirm this?